node/heartbeat/monitor.go

```diff
@@ -169,21 +170,15 @@ func (m *Monitor) ProcessReceivedMessage(message p2p.MessageP2P) error {
 
 	//message is validated, process should be done async, method can return nil
 	go m.addHeartbeatMessageToMap(hbRecv)
-	go func() {
-		m.mutHeartbeatMessages.Lock()
-		defer m.mutHeartbeatMessages.Unlock()
 
-		m.computeAllHeartbeatMessages()
-	}()
+	go m.computeAllHeartbeatMessages()
 
 	return nil
 }
 
 func (m *Monitor) addHeartbeatMessageToMap(hb *Heartbeat) {
-	m.mutHeartbeatMessages.Lock()
-	defer m.mutHeartbeatMessages.Unlock()
-
 	pubKeyStr := string(hb.Pubkey)
+	m.mutHeartbeatMessages.Lock()
 	hbmi, ok := m.heartbeatMessages[pubKeyStr]
 	if hbmi == nil || !ok {
 		var err error
```

Comment:
add m.mutheartbeatmessages.unlock to line 188

Reply:
fixed